x-pack/legacy/plugins/actions/server/builtin_action_types/slack.ts

```diff
@@ -35,9 +35,9 @@ const ParamsSchema = schema.object({
 // action type definition
 
 // customizing executor is only used for tests
-export function getActionType({ executor }: { executor?: ExecutorType } = {}): ActionType {
-  if (executor == null) executor = slackExecutor;
-
+export function getActionType(
+  { executor }: { executor: ExecutorType } = { executor: slackExecutor }
```

Comment:
fyi: this is a kind of awful hack I did on the slack action (1st real action) to mock the service call - I have a much better approach to this in the pagerduty action, using jest import mocking.  At some point if we need to do more work on slack, we should probably switch to that pattern.

Reply:
Fair, though I actually think there might be an even cleaner approach.
The tests that use the registry to unit test an action seem off to me - they're creating unnecessary coupling between the action's tests and the registry's implementation.
If we test the executor directly, and test the _registration_ as part of the registry's integration tests- that seems less likely to force unnecessary changes further down the line. :)